src/software/thunderscope/field/world_layer.py

```diff
@@ -213,10 +213,25 @@ def draw_team(self, painter, color, team: Team):
 
         for robot in team.team_robots:
 
+            robot_id_font = painter.font()
+            robot_id_font.setPointSize(ROBOT_MAX_RADIUS / 7)
+            # setting a black background to keep ID visible over yellow robot
+            robot_id_text = pg.TextItem(
+                html='<span style="color: #FFF; background-color: #000">'
+                + str(robot.id)
+                + "</span>"
+            )
+            robot_id_text.setFont(robot_id_font)
+            robot_id_text.setPos(
+                (robot.current_state.global_position.x_meters * MM_PER_M)
+                - ROBOT_MAX_RADIUS,
+                robot.current_state.global_position.y_meters * MM_PER_M,
+            )
+            robot_id_text.setParentItem(self)
```

Comment:
tested it out locally, this call in this loop really bogs down thunderscope because we are adding a new text item for every robot on every ticket

we should rather just create text item per robot and track it in a map (robot id -> text item). If a new robot id is detected, just add it to the map. only call setParentItem if the robot id is new

In this loop, if the robot id exists, just update the position of the text item.